python/tank/util/shotgun/connection.py

```diff
@@ -197,6 +197,14 @@ def _raise_missing_key(key):
     if not config_data.get("api_script") and config_data.get("api_key"):
         _raise_missing_key("api_script")
 
+    # Manage configurations containing environment variables
+    for key in config_data:
+        if isinstance(config_data.get(key, None), str):
+            config_data[key] = os.path.expandvars(config_data[key])
+            # Id keys should be integers
```

Comment:
I don't believe shotgun.yml has values suffixed with `id`. What did you have in mind here? I suspect this should be removed.

Reply:
You're right. Fixed!